internal/xdscache/v3/listener.go

```diff
@@ -502,7 +502,11 @@ func (c *ListenerCache) OnChange(root *dag.DAG) {
 
 				filters = envoy_v3.Filters(cm)
 
-				alpnProtos = envoy_v3.ProtoNamesForVersions(cfg.DefaultHTTPVersions...)
+				if len(vh.HttpVersions) != 0 {
+					alpnProtos = vh.HttpVersions
```

Comment:
In this case where the vhost specifies HTTP versions, I think we can also use the virtual host's HTTP versions for the HTTPConnectionManager's codec, above, since it's specific to the vhost.

Reply:
This comment still needs to be addressed

Reply:
Sure. I'll do when there's an agreement on [this](https://github.com/projectcontour/contour/pull/5802#discussion_r1443650964).